soh/src/code/z_eff_blure.c

```diff
@@ -196,11 +205,203 @@ void EffectBlure_Destroy(void* thisx) {
 s32 EffectBlure_Update(void* thisx) {
     EffectBlure* this = (EffectBlure*)thisx;
     s32 i;
-    s16 RedColor;
-    s16 GreenColor;
-    s16 BlueColor;
-    s16 TrailDuration;
-    Color_RGB8 Trails_col = CVar_GetRGB("gTrailCol", Trails_Color_ori);
+
+    Color_RGB8 SwordTopCol = CVar_GetRGB("gSwordTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 SwordBottomCol = CVar_GetRGB("gSwordTrailBottomCol", TrailsColorOriginal);
+    Color_RGB8 BoomStartCol = CVar_GetRGB("gBoomTrailStartCol", BoomColorOriginal);
+    Color_RGB8 BoomEndCol = CVar_GetRGB("gBoomTrailEndCol", BoomColorOriginal);
+    Color_RGB8 BombchuCol = CVar_GetRGB("gBombTrailCol", BombColorOriginal);
+    Color_RGB8 KSwordTopCol = CVar_GetRGB("gKSwordTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 KSwordBottomCol = CVar_GetRGB("gKSwordTrailBottomCol", TrailsColorOriginal);
+    Color_RGB8 MSwordTopCol = CVar_GetRGB("gMSwordTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 MSwordBottomCol = CVar_GetRGB("gMSwordTrailBottomCol", TrailsColorOriginal);
+    Color_RGB8 BSwordTopCol = CVar_GetRGB("gBSwordTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 BSwordBottomCol = CVar_GetRGB("gBSwordTrailBottomCol", TrailsColorOriginal);
+    Color_RGB8 StickTopCol = CVar_GetRGB("gStickTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 StickBottomCol = CVar_GetRGB("gStickTrailBottomCol", TrailsColorOriginal);
+    Color_RGB8 HammerTopCol = CVar_GetRGB("gHammerTrailTopCol", TrailsColorOriginal);
+    Color_RGB8 HammerBottomCol = CVar_GetRGB("gHammerTrailBottomCol", TrailsColorOriginal);
+
+    if ((CVar_GetS32("gUseTrailsCol", 0) != 0) && (this->trailType != 0)) {
+        switch (this->trailType) { //there HAS to be a better way to do this.
```

Comment:
this would probably feel less bad if trailType was an enum, then we'd have something like
`case TRAIL_TYPE_SWORD:`
`case TRAIL_TYPE_BOOMERANG:`
etc.

Reply:
To be fair, I'd love to know how to do that. I just kinda...don't.

Reply:
good example here https://github.com/HarbourMasters/Shipwright/blob/f6fa3c74c7cdbca50c9218f49ac36bc0d62f9e3b/soh/src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.h#L9-L13

probably should put that in `z64effect` and then instead of using `u8` for `trailType` you could use `TrailType` (or whatever you end up naming the enum) as the type